Cocoa powder and instant cocoa powder milk

Kray4000 opened the pull request:
#### Summary
Content "Cocoa powder, Milk cocoa powder"


#### Purpose of change

Looked for some cocoa in DDA and can't find it. That one of the most common drink after coffe and tea.

#### Describe the solution

Added `"cocoa_powder"` and `"cocoa_powder_milk"`, connect it to recipes, added it to itemgroups.

#### Describe alternatives you've considered

1. Ask someone to add it

2. Life in Cataclysm without cocoa 

#### Testing


#### Additional context

1. My first time when I adding item to itemgroups, can be mistakes
2. Copied from milk powder (30g) - it might be better to make other dosages

Kray4000 committed:
Cocoa powder and instant cocoa powder milk

Kray4000 committed:
Update food.json

added missed comma

Kray4000 committed:
Update domestic.json

comma again

Kray4000 committed:
Update domestic.json

- ,

Kray4000 committed:
Update food.json

- ,

Kray4000 committed:
Update recipe_food.json

Kray4000 committed:
Update recipe_food.json

Kray4000 committed:
Update drinks.json

Kray4000 committed:
Update drinks.json

Kray4000 committed:
Water_dissolve and material fix

Kray4000 committed:
Update drinks.json

Kray4000 committed:
Update data/json/items/comestibles/other.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Kray4000 committed:
Update data/json/itemgroups/Food/food.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Kray4000 committed:
Update data/json/itemgroups/SUS/domestic.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Kray4000 committed:
Update data/json/items/comestibles/other.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Kray4000 committed:
Update data/json/itemgroups/Locations_MapExtras/locations_commercial.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Kray4000 committed:
Update data/json/items/comestibles/other.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Kray4000 committed:
Update data/json/items/comestibles/other.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

GuardianDll committed:
fix some errors, lint

Kray4000 committed:
New milk_choc recipe with cocoa_powder_milk

Kray4000 committed:
Update drinks.json

Kray4000 committed:
Update recipe_food.json

Kray4000 committed:
Update drinks.json

Kray4000 committed:
Update drinks.json

Kray4000 committed:
Update drinks.json

Kray4000 committed:
Update drinks.json

TheShadowFerret commented on data/json/recipes/recipe_food.json:
```diff
@@ -3236,6 +3236,19 @@
+    "skill_used": "cooking",
+    "time": "2 m",
+    "autolearn": true,
+    "//": "you're just dissolve the milk cocoa powder in the water. That take some time because cocoa powder have some fat. Therefore, you need to mix well",
```
```suggestion
    "//": "You're just dissolving the milk cocoa powder into the water. That takes some time because cocoa powder has some fat. Therefore, you need to mix well",
```
~~Typos in the player-invisible section reeee~~

Kray4000 committed:
Update data/json/recipes/recipe_food.json

Co-authored-by: TheShadowFerret <99621099+TheShadowFerret@users.noreply.github.com>

Kray4000 committed:
Update drinks.json

Kray4000 committed:
Update drinks.json

Kray4000 committed:
Update drinks.json (trying to fix Json Validation error)

Error: 
Run make style-all-json-parallel RELEASE=1
g++ -Os -Wodr -Werror -Wall -Wextra -Wformat-signedness -Wlogical-op -Wmissing-declarations -Wmissing-noreturn -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -Wpedantic -Wsuggest-override -Wunused-macros -Wzero-as-null-pointer-constant -Wno-unknown-warning-option -Wno-dangling-reference -Wno-c++20-compat -Wredundant-decls -g1 -fsigned-char -std=c++17 -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600 -MMD -MP -DCATA_IN_TOOL -Itools/format -Isrc -isystem src/third-party \
  tools/format/format.cpp tools/format/format_main.cpp src/wcwidth.cpp src/json.cpp -o tools/format/json_formatter.cgi
find data -name "*.json" -print0 | xargs -0 -L 1 -P $(nproc) tools/format/json_formatter.cgi
Has been linted : data/json/recipes/food/drinks.json
Please read doc/JSON/JSON_STYLE.md
make: *** [Makefile:1387: style-all-json-parallel] Error 123
Error: Process completed with exit code 2.

Kray4000 committed:
Update other.json (deleting volume old duplicate)

Kray4000 commented:
Seems like some recepts like 
```
  {
    "type": "recipe",
    "activity_level": "NO_EXERCISE",
    "result": "sweet_milk_fortified",
    "category": "CC_FOOD",
    "subcategory": "CSC_FOOD_DRINKS",
    "skill_used": "cooking",
    "time": "6 m",
    "autolearn": true,
    "flags": [ "BLIND_HARD" ],
>     "components": [
>       [ [ "protein_powder", 1 ] ],
>       [ [ "sugar_standard_half", 1, "LIST" ] ],
>       [ [ "milk_standard", 1, "LIST" ], [ "lf_milk_standard", 1, "LIST" ] ]

    ]
  },
```
 ... have wrong Json style and it is great example of new contributors errors , because we often use copy-forms. And this, of course, will leads to Json Validation errors. Was json style changed somewhere in development process? Or this is "ok, don't mind this error"? Should then json Validation be updated?

Kray4000 committed:
Update other.json

Kray4000 committed:
Update drinks.json

 (There already hot_chocolate recipe)

Kray4000 committed:
Update recipe_food.json

Kray4000 committed:
Update recipe_food.json

Kray4000 committed:
Update drinks.json

Kray4000 committed:
Update drinks.json

Kray4000 committed:
Update recipe_food.json

Kray4000 committed:
Update recipe_food.json

Kray4000 committed:
Update drinks.json

Kray4000 committed:
Update drinks.json

Kray4000 committed:
Update recipe_food.json (Сhanging the amount of powder to suit the chocolate weight requirements)

Сhanging the amount of powder to suit the chocolate weight requirements (chocolate 43 g, powder is 30)
So in Mex_chocolate need at least 2 choc. bars (86 g) - then will be 3 powder charges (90 g)

Kray4000 commented:
[57dbac6](https://github.com/CleverRaven/Cataclysm-DDA/pull/80227/commits/57dbac68c82cf677dba1dd15a62b9a6a1a1ab962)
Description is wrong. Must be: `1 bar - 43 g so 2 powder charges - 60 g because 1 is 30 g and it is too small portion`

Kray4000 committed:
Update other.json

Kray4000 committed:
Update other.json

Kray4000 committed:
Update other.json

Kray4000 committed:
Update other.json

Kray4000 committed:
Update collections_domestic.json

Kray4000 committed:
Update food.json

Kray4000 committed:
Update food.json

Kray4000 committed:
Update collections_domestic.json

Kray4000 committed:
Update data/json/itemgroups/collections_domestic.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Kray4000 committed:
Update data/json/itemgroups/collections_domestic.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Kray4000 committed:
Update data/json/itemgroups/collections_domestic.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Kray4000 committed:
Update domestic.json

Kray4000 committed:
Update domestic.json

Kray4000 committed:
Update domestic.json

Kray4000 committed:
Update collections_domestic.json

Kray4000 committed:
Update domestic.json

Kray4000 committed:
Update collections_domestic.json

Kray4000 committed:
Update collections_domestic.json (fix volume)

Kray4000 committed:
Update collections_domestic.json

Kray4000 committed:
Deleting can_food_big variant

Kray4000 committed:
Update data/json/itemgroups/collections_domestic.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Kray4000 committed:
Update data/json/itemgroups/collections_domestic.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Kray4000 commented:
I'm tired to reroll CI tests in this PR, it survive draft check but can't pass after been ready to merge, after couple attempts, So I just don't know what problem is. Maybe if I knew why it happening I'll ask about kick, but I don't know where problem is, in my code or in CI tests because every time there another reason , and ask about every error will be annoying for people.

GuardianDll commented:
just needed to update your branch, it didn't have the commit where CI test was fixed

Kray4000 committed:
Merge branch 'CleverRaven:master' into Cacao-powder
